broadinstitute/single_cell_portal_core
# Cell Facet Filter Component (SCP-5215)

Initial work for the cell facet filtering UI work. 

This builds on the library work for cell facet filtering and adds controls to the UI. Starting with 3 dropdown menus listing the 5 prioritized annotations/facets returned by the library built [here](https://github.com/broadinstitute/single_cell_portal_core/pull/1862). Pre-populated but allows for changing facet choice. Allowing additional, or removal entirely of facets is work to be done in [TICKET](https://broadworkbench.atlassian.net/browse/SCP-5297?atlOrigin=eyJpIjoiNzY4NGRlOWIyNzU2NDQzNWI1NmZmYWJhMTE0OGUxYTciLCJwIjoiaiJ9). This is gated by a feature flag so won't be GA until the team is satisfied will not be visible to the users.


Todos that still need to be done, I can address in this PR or as subsequent work:
- Loading spinners are not aligned properly
- Needs tests
- Some general UI touch ups like colors, border edge rounding, spacing, etc. 
-
I wanted to get this out for review ASAP but recognize these ^ things still need to be addressed.


To test:
- pull this branch
- run the migration to ensure the new feature flag is available
- start up a local portal and log in
- flip the feature flag to true for yourself from the feature flags dashboard
- navigate to a study
- you should see the button for facet filtering, similar to the button for DE in the options panel
- clicking the button will show the facets and their filters, clicking the check boxes next to the filters will add or remove points from the plot as appropriate
- you can change the facets using the drop downs, you can also change the "color by" annotation using that dropdown
- clicking the back arrow to return to the default options panel should reset the plot to showing all the points


https://github.com/broadinstitute/single_cell_portal_core/assets/54322292/368cbcc9-d153-47e7-bd4d-a58ffd5ad718


https://github.com/broadinstitute/single_cell_portal_core/assets/54322292/28b1ff43-8a80-455d-b54d-633693b035f9
